### PR TITLE
keycloak migration

### PR DESCRIPTION
**Pull request by bdolor**
This PR closes issue: [DM-950]

Includes tests? [N]
Updated docs? [Y]

Proposed changes:
- migrates to new KeyCloak instance

### CODE REVIEW

File: docs/openshift.md [hunk 1]
```diff
@@ -67,9 +67,29 @@ oc process -f openshift/templates/database/patroni-digmkt-deploy.yaml -p TAG_NAM
...
+
+```yaml
+KEYCLOAK_CLIENT_ID="<resource>"
+KEYCLOAK_CLIENT_SECRET="<secret>"
+KEYCLOAK_URL="<url>"
+KEYCLOAK_REALM="<realm>"
+```
 
-To password protect the dev and test namespace deployments, replace `BASIC_AUTH_USERNAME` and `BASIC_AUTH_PASSWORD_HASH` with the basic auth credentials desired. Leaving these parameters off the deployment command will deactivate login. The hashed password can be generated using the npm library bcrypt `bcrypt.hash('<password>',10)`. (NOTE:  This login is unrelated to keycloak authentication.)
+**Optional:** To password protect the dev and test namespace deployments, replace `BASIC_AUTH_USERNAME` and `BASIC_AUTH_PASSWORD_HASH` with the basic auth credentials desired. Leaving these parameters off the deployment command will deactivate login. The hashed password can be generated using the npm library bcrypt `bcrypt.hash('<password>',10)`. (NOTE:  This login is unrelated to keycloak authentication.)
```
Comment (IanFonzie) [inline docs/openshift.md:92]: Does this need to be generated on every deployment to dev/test?

File: src/back-end/lib/routers/auth.ts [hunk 2]
```diff
@@ -382,14 +382,47 @@ async function establishSessionWithClaims(
...
   let userType: UserType;
-  const identityProvider = getString(claims, "loginSource");
+  const identityProvider = getString(claims, "identity_provider");
   switch (identityProvider) {
-    case GOV_IDP_SUFFIX.toUpperCase():
+    case GOV_IDP_SUFFIX:
       userType = UserType.Government;
       break;
-    case VENDOR_IDP_SUFFIX.toUpperCase():
+    case VENDOR_IDP_SUFFIX:
```
Comment (IanFonzie) [inline src/back-end/lib/routers/auth.ts:425]: Are the `*_IDP_SUFFIX`es guaranteed to be lowercase?

**Review by IanFonzie — COMMENTED** [review]
Looks good, had a few questions:

File: docs/openshift.md (hunk 1 above)
Comment (bdolor) [reply docs/openshift.md:92]: Deployment will occur if they are left out. They are currently optional parameters that will not result in deployment failure if left out.

File: src/back-end/lib/routers/auth.ts (hunk 2 above)
Comment (bdolor) [reply src/back-end/lib/routers/auth.ts:425]: it's guaranteed on the application side via https://github.com/bcgov/digital_marketplace/blob/development/src/shared/config.ts#L17 and on the SSO side, it's changed from coming in as: 

```
{
...
  loginSource: 'IDIR'
}
``` 

to: 

```
{
...
identity_provider: 'idir'
}

**Comment by bdolor** [comment]
replaced with hotfix: https://github.com/bcgov/digital_marketplace/pull/338

**Pull request closed without merging**